traffic_ops/testing/api/v3/cachegroupsdeliveryservices_test.go

```diff
@@ -0,0 +1,137 @@
+package v3
+
+/*
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"testing"
+)
+
+func TestCacheGroupsDeliveryServices(t *testing.T) {
+	WithObjs(t, []TCObj{CDNs, Types, Tenants, Parameters, Profiles, Statuses, Divisions, Regions, PhysLocations, CacheGroups, Servers, DeliveryServices, CacheGroupsDeliveryServices}, func() {})
+}
+
+// TODO this is the name hard-coded in the create servers test; change to be dynamic
+// TODO this test assumes that a CDN named "cdn1" exists, has at least one Delivery Service, and also
+// assumes that ALL SERVERS IN "cachegroup3" ARE EDGE-TIER CACHE SERVERS IN "cdn1". If that EVER changes,
+// this WILL break.
+const TestEdgeServerCacheGroupName = "cachegroup3"
+
+func CreateTestCachegroupsDeliveryServices(t *testing.T) {
+	dss, _, err := TOSession.GetDeliveryServiceServers()
+	if err != nil {
+		t.Fatalf("cannot GET DeliveryServiceServers: %v", err)
+	}
+	if len(dss.Response) > 0 {
+		t.Fatalf("cannot test cachegroups delivery services: expected no initial delivery service servers, actual %v", len(dss.Response))
+	}
+
+	dses, _, err := TOSession.GetDeliveryServices()
```

Comment:
I checked and it was there. Also ran the tests - but at any rate I've rebased again since then and #4670 is merged so tests should both compile and pass

Reply:
and also RE: this file in particular - should be recopied now